src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -128,6 +133,126 @@ internal partial class Binder
+            try
+            {
+                var unboundObjectCreation = (UnboundObjectCreationExpression)source;
+
+                // PROTOTYPE(target-typed-new): Reconstruct AnalyzedArguments to avoid binding twice
+                arguments.Arguments.AddRange(unboundObjectCreation.Arguments);
+                if (!unboundObjectCreation.ArgumentRefKindsOpt.IsDefault)
+                {
+                    arguments.RefKinds.AddRange(unboundObjectCreation.ArgumentRefKindsOpt);
+                }
+
+                if (!unboundObjectCreation.ArgumentNamesOpt.IsDefault)
+                {
+                    arguments.Names.AddRange(unboundObjectCreation.ArgumentNamesOpt);
+                }
+
+                if (ReportBadTargetType(syntax, type, diagnostics))
+                {
+                    return MakeBadExpressionForObjectCreation(node, destination, boundInitializerOpt, arguments);
+                }
+
+                BoundExpression result;
+                switch (type.TypeKind)
+                {
+                    case TypeKind.Struct:
+                    case TypeKind.Class:
+                        result = BindClassCreationExpression(
+                            node,
+                            typeName: type.Name,
+                            typeNode: node,
+                            type: (NamedTypeSymbol)type,
+                            arguments,
+                            diagnostics,
+                            boundInitializerOpt,
+                            forTargetTypedNew: true);
+                        break;
+
+                    case TypeKind.TypeParameter:
+                        result = BindTypeParameterCreationExpression(
+                            node,
+                            typeParameter: (TypeParameterSymbol)type,
+                            arguments,
+                            boundInitializerOpt,
+                            diagnostics);
+                        break;
+
+                    default:
+                        throw ExceptionUtilities.UnexpectedValue(type.Kind);
+                }
+
+                if (destination.IsNullableType())
+                {
+                    // We manually create an ImplicitNullable conversion
+                    // if the destination is nullable, in which case we
+                    // target the underlying type e.g. `S? x = new();`
+                    // is actually identical to `S? x = new S();`.
+                    result = new BoundConversion(
+                        syntax,
+                        result,
+                        conversion: new Conversion(ConversionKind.ImplicitNullable, Conversion.IdentityUnderlying),
+                        @checked: false,
+                        explicitCastInCode: isCast,
+                        constantValueOpt: null, // A "target-typed new" would never produce a constant.
+                        type: destination);
```

Comment:
This should be explicit in the spec, if it isn't already, as the following corresponding code is accepted today:
``` c#
        const int N = new int();
```
I think there is a rule that a value type's default constructor can't be used; that is what causes this (commented fact) to be true.